core/common/src/main/java/alluxio/collections/NonUniqueFieldIndex.java

```diff
@@ -98,4 +111,70 @@ public T getFirst(Object value) {
     Set<T> all = mIndexMap.get(value);
     return all == null ? null : Iterables.getFirst(all, null);
   }
+
+  @Override
+  public Iterator<T> iterator() {
+    return new NonUniqueFieldIndexIterator();
+  }
+
+  @Override
+  public int size() {
+    int totalSize = 0;
+    for (ConcurrentHashSet<T> innerSet : mIndexMap.values()) {
```

Comment:
Because `size()` is not heavily used and make this O(1) could bring memory overhead. So I leave it as origin. But I can change it to O(1) now.